fix issues causing test failures with Python 3.6
partially inspired by https://bugs.archlinux.org/task/52409
author: Kenneth Hoste (HPC-UGent)
--- openbabel-2.4.1/scripts/python/openbabel.py.orig	2019-02-13 17:59:40.515899728 +0100
+++ openbabel-2.4.1/scripts/python/openbabel.py	2019-02-13 18:00:22.775794400 +0100
@@ -3,7 +3,11 @@
   try:
     import dl
   except ImportError:
-    import DLFCN as dl
+    try:
+        import DLFCN as dl
+    except ImportError:
+        # fallback for Python 3.x
+        import os as dl
   sys.setdlopenflags(sys.getdlopenflags() | dl.RTLD_GLOBAL)
 # This file was automatically generated by SWIG (http://www.swig.org).
 # Version 3.0.10
--- openbabel-2.4.1/test/testpdbformat.py.orig	2016-03-02 22:09:51.000000000 +0100
+++ openbabel-2.4.1/test/testpdbformat.py	2019-02-13 17:50:27.917372676 +0100
@@ -24,12 +24,12 @@
 
     def testInsertionCodes(self):
         """
-	Testing a PDB entry with insertion codes to distinguish residues
-	upon conversion to FASTA.
+    Testing a PDB entry with insertion codes to distinguish residues
+    upon conversion to FASTA.
         """
         self.canFindExecutable("babel")
 
-	self.entryPDBwithInsertioncodes="""ATOM    406  N   VAL L  29      58.041  17.797  48.254  1.00  0.00           N  
+        self.entryPDBwithInsertioncodes="""ATOM    406  N   VAL L  29      58.041  17.797  48.254  1.00  0.00           N  
 ATOM    407  CA  VAL L  29      57.124  18.088  47.170  1.00  0.00           C  
 ATOM    408  C   VAL L  29      55.739  17.571  47.538  1.00  0.00           C  
 ATOM    409  O   VAL L  29      55.535  16.362  47.550  1.00  0.00           O  
@@ -100,9 +100,9 @@
 ATOM    474  HE2 TYR L  32      48.145  19.172  44.648  1.00  0.00           H  
 ATOM    475  HH  TYR L  32      46.462  17.658  44.280  1.00  0.00           H  
 """
-	output, error = run_exec(self.entryPDBwithInsertioncodes,
-				     "babel -ipdb -ofasta")
-	self.assertEqual(output.rstrip().rsplit("\n",1)[1], "VSSSY")
+        output, error = run_exec(self.entryPDBwithInsertioncodes,
+                                 "babel -ipdb -ofasta")
+        self.assertEqual(output.rstrip().rsplit("\n",1)[1], "VSSSY")
 
 if __name__ == "__main__":
     testsuite = []
